### DISCUSSION TITLE
June 2022 Release (version 1.19.3) - Menu, Native Keyboard, ignoreBlur

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#787] [Announcements]
# Script Kit 1.19.3 Released 🎉

Download here: https://www.scriptkit.com/
Direct downloads:
* Intel: https://github.com/johnlindquist/kitapp/releases/download/v1.19.3/Kit-1.19.3.dmg
* m1: https://github.com/johnlindquist/kitapp/releases/download/v1.19.3/Kit-1.19.3-arm64.dmg

## Features

* Moved all notifications to menu colored “dots” (red, orange, green)
    * In the future, the user will be able to set these notifications as well
* Completely re-vamped menu for common options
* Implemented an “auth helpers” when user does actions that require “accessibility permissions” (Snippets, clipboard history, setSelectedText, etc)
* `ignoreBlur` allows window to go behind other windows and stay open
* `node` is now stored in `~/.knode` (instead of ~/.kit/node) to allow npx to work in the terminal
* `node` version set to v16.14.2. Version is now synced with Kit.app which resolves conflicts with native packages in scripts
* Keyboard actions (copy/paste/type) have moved from applescript to native code. Snippets, setSelectedText, etc should now feel as “instant” as possible.
* You can now `await hide()` for when you need to make sure the prompt is hidden before continuing the script. This was necessary since the new keyboard actions were so fast.
* Moved the script sharing auth flow to a widget
* Internal: Can now set the state of Kit.app from a script to help with debugging

## Fixes

* App launcher failed to parse malformed App plist icons
* Editor whitespace collapsing on HiDPI screens
* Touchbar key while prompt open would cause crash
* Bin files sometimes didn’t regenerate properly when re-launching the app
* Updater issues
* Background UI not updating when user manually terminates process
* Performance: Moved the file watcher to a spawned process that sends events to the App

## More to come...

Finally moved into the new house and settled in. Personal life was too hectic to stream much or do release notes on past couple releases. Expect more streaming, sharing scripts, promotion, and news from me. Cheers! 🥂

_Here's a preview of the new menu and an example of a dot notification_

![The new dot notifications and re-vamped menu](https://cdn.discordapp.com/attachments/963905444823318578/988844882665803926/unknown.png)

**Comment by 07810365582** [comment]
ppsspp
